File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        def subSum(low,high):
            if low < high:
                mid=(low+high)//2
                max1=subSum(low,mid)
                max2=subSum(mid+1,high)
                max3=nums[mid]
                s1=0
                for j in range(mid,low-1,-1):
                    s1+=nums[j]
                    max3=max(s1,max3)
                s1=0
                m3=nums[mid+1]
                for j in range(mid+1,high+1):
                    s1+=nums[j]
                    m3=max(s1,m3)
                max3+=m3
                return max(max1,max2,max3)
            elif high==low:
                return nums[low]
            
        return subSum(0,len(nums)-1)

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int maxSum = nums[0];
        int curSum=nums[0];
        int l=0;
        int j;
        for(j=1; j<nums.length ;j++){
            
            while(l < nums.length && curSum <0){
                curSum-=nums[l];
                l++;
            }
            curSum+=nums[j];
            maxSum=Math.max(maxSum,curSum);

        }
        return maxSum;

        
    }
}